src/internal/event.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+/**
+ * This file contains the underlying implementations for exposed API surface in
+ * the {@link api/event}. By moving the methods out into a separate file,
+ * other namespaces and processes can access these methods without depending on the entire
+ * account namespace and without having a dependency cycle error.
+ */
+
+import { AptosConfig } from "../api/aptos_config";
+import { AnyNumber, GetEventsByCreationNumberResponse, HexInput } from "../types";
+import { GetEventsByCreationNumberQuery } from "../types/generated/operations";
+import { GetEventsByCreationNumber } from "../types/generated/queries";
+import { queryIndexer } from "./general";
+
+/**
+ * Get events by creation number and the address
+ *
+ * @param args.aptosConfig - The aptos config
+ * @param args.address - The account address
+ * @param args.creationNumber - The event creation number
+ * @returns Promise<GetEventsByCreationNumberResponse>
+ */
+export async function getEventsByCreationNumber(args: {
+  aptosConfig: AptosConfig;
+  address: HexInput;
+  creationNumber: AnyNumber;
+}): Promise<GetEventsByCreationNumberResponse> {
+  const { aptosConfig, address, creationNumber } = args;
+  const graphqlQuery = {
+    query: GetEventsByCreationNumber,
+    variables: { address, creationNumber },
```

Comment:
```suggestion
    variables: { Hex.fromHexInput({ hexInput: address }).toString(), creationNumber },
```

maybe add some input validation?  oof I hate the object as params for this method....

Reply:
Good catch!